From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rezoss <rezoss79@users.noreply.github.com>
Date: Sun, 21 Jun 2026 00:00:00 +0200
Subject: [PATCH] Handle missing Kakao Navi provider

Kakao parking integration depends on the Kakao Navi Now Brief provider,
which is not always present on this port. Treat missing login state as
false and make provider-backed parking flows inert instead of registering
observers on an authority that PackageManager cannot resolve.

---
 .../utils/kakaonavi/KakaoParkingContentProviderHelper$getIsLogin$2$1.smali | 9 +++++----
 .../remote/parking/KakaoParkingDataSource$getIsLogin$2$1.smali             | 11 +++++------
 .../documentgenerator/data/remote/parking/KakaoParkingDataSource.smali     | 14 ++++++++++++++
 3 files changed, 24 insertions(+), 10 deletions(-)

diff --git a/smali_classes14/com/samsung/android/smartsuggestions/feature/aisuggestion/briefsettings/utils/kakaonavi/KakaoParkingContentProviderHelper$getIsLogin$2$1.smali b/smali_classes14/com/samsung/android/smartsuggestions/feature/aisuggestion/briefsettings/utils/kakaonavi/KakaoParkingContentProviderHelper$getIsLogin$2$1.smali
--- a/smali_classes14/com/samsung/android/smartsuggestions/feature/aisuggestion/briefsettings/utils/kakaonavi/KakaoParkingContentProviderHelper$getIsLogin$2$1.smali
+++ b/smali_classes14/com/samsung/android/smartsuggestions/feature/aisuggestion/briefsettings/utils/kakaonavi/KakaoParkingContentProviderHelper$getIsLogin$2$1.smali
@@ -236,13 +236,15 @@
     goto :goto_0

     :cond_0
-    iget-object v0, p0, Lcom/samsung/android/smartsuggestions/feature/aisuggestion/briefsettings/utils/kakaonavi/KakaoParkingContentProviderHelper$getIsLogin$2$1;->$contentResolver:Landroid/content/ContentResolver;
+    iget-object p0, p0, Lcom/samsung/android/smartsuggestions/feature/aisuggestion/briefsettings/utils/kakaonavi/KakaoParkingContentProviderHelper$getIsLogin$2$1;->$continuation:Lkotlinx/coroutines/g;

-    iget-object p0, p0, Lcom/samsung/android/smartsuggestions/feature/aisuggestion/briefsettings/utils/kakaonavi/KakaoParkingContentProviderHelper$getIsLogin$2$1;->$uri:Landroid/net/Uri;
+    sget-object v0, Ljava/lang/Boolean;->FALSE:Ljava/lang/Boolean;

-    const/4 v1, 0x1
+    invoke-static {v0}, Lkotlin/Result;->constructor-impl(Ljava/lang/Object;)Ljava/lang/Object;

-    invoke-virtual {v0, p0, v1, p1}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
+    move-result-object p1
+
+    invoke-interface {p0, p1}, Lkotlin/coroutines/c;->resumeWith(Ljava/lang/Object;)V

     :goto_0
     sget-object p0, Lkotlin/r;->a:Lkotlin/r;
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource$getIsLogin$2$1.smali b/smali_classes15/com/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource$getIsLogin$2$1.smali
--- a/smali_classes15/com/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource$getIsLogin$2$1.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource$getIsLogin$2$1.smali
@@ -227,17 +227,15 @@
     goto :goto_0

     :cond_0
-    iget-object v0, p0, Lcom/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource$getIsLogin$2$1;->this$0:Lcom/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource;
+    iget-object p0, p0, Lcom/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource$getIsLogin$2$1;->$continuation:Lkotlinx/coroutines/g;

-    invoke-static {v0}, Lcom/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource;->access$getContentResolver$p(Lcom/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource;)Landroid/content/ContentResolver;
+    sget-object v0, Ljava/lang/Boolean;->FALSE:Ljava/lang/Boolean;

-    move-result-object v0
+    invoke-static {v0}, Lkotlin/Result;->constructor-impl(Ljava/lang/Object;)Ljava/lang/Object;

-    iget-object p0, p0, Lcom/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource$getIsLogin$2$1;->$uri:Landroid/net/Uri;
+    move-result-object p1

-    const/4 v1, 0x1
-
-    invoke-virtual {v0, p0, v1, p1}, Landroid/content/ContentResolver;->registerContentObserver(Landroid/net/Uri;ZLandroid/database/ContentObserver;)V
+    invoke-interface {p0, p1}, Lkotlin/coroutines/c;->resumeWith(Ljava/lang/Object;)V

     :goto_0
     sget-object p0, Lkotlin/r;->a:Lkotlin/r;
diff --git a/smali_classes15/com/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource.smali b/smali_classes15/com/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource.smali
--- a/smali_classes15/com/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource.smali
+++ b/smali_classes15/com/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource.smali
@@ -768,6 +768,14 @@
     .annotation build Lorg/jetbrains/annotations/NotNull;
     .end annotation

+    sget-object v0, Ljava/lang/Boolean;->FALSE:Ljava/lang/Boolean;
+
+    invoke-static {v0}, Lkotlinx/coroutines/flow/i;->z(Ljava/lang/Object;)Lkotlinx/coroutines/flow/g;
+
+    move-result-object v0
+
+    return-object v0
+
     const-string v0, "content://com.kakao.navi.nowbrief.provider/is_login"

     invoke-static {v0}, Landroid/net/Uri;->parse(Ljava/lang/String;)Landroid/net/Uri;
@@ -818,6 +826,12 @@
     .annotation build Lorg/jetbrains/annotations/NotNull;
     .end annotation

+    invoke-static {}, Lkotlinx/coroutines/flow/i;->r()Lkotlinx/coroutines/flow/g;
+
+    move-result-object p0
+
+    return-object p0
+
     invoke-direct {p0, p3, p4}, Lcom/samsung/android/smartsuggestions/feature/aisuggestion/documentgenerator/data/remote/parking/KakaoParkingDataSource;->formatUtcTimeString(J)Ljava/lang/String;

     move-result-object p3